src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -340,11 +348,23 @@ export class EditDataGridPanel extends GridParentComponent {
 
 		// At the end of a successful cell select, update the currently selected cell
 		cellSelectTasks = cellSelectTasks.then(() => {
+			this.updateEnabledState(true);
 			self.setCurrentCell(row, column);
+			self.focusCell(row, column);
 		});
 
 		// Cap off any failed promises, since they'll be handled
-		cellSelectTasks.catch(() => { });
+		cellSelectTasks.catch(() => {
+		});
+	}
+
+	/**
+	 * Disables editing the grid temporarily when clicking on a cell (to allow for any processing tasks to be finished first such as adding a new row).
+	 * @param state The variable telling whether to enable selection of the table cells or not.
+	 */
+	private updateEnabledState(state: boolean): void {
+		//Need to suppress rerendering to avoid infinite loop when changing new row.
+		this.table.grid.setOptions({ editable: state }, true);
```

Comment:
you have to include other options here that were used when the table is created: https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts#L835